### DISCUSSION TITLE
Fail to compile velox in 10min example code

### DISCUSSION THREAD

**Discussion by yizhuoliang** [facebookincubator/velox#11597] [Q&A] [answered]
I build velox with `make minimal`, and trying to build the velox in 10 minutes example. But got the compile error
```
cc@mc-1:~/vtest$ g++ -o ten -I/home/cc/velox -L/home/cc/velox/_build/release/velox ten.cc
In file included from /home/cc/velox/velox/buffer/Buffer.h:26,
                 from /home/cc/velox/velox/vector/BaseVector.h:29,
                 from /home/cc/velox/velox/vector/ComplexVector.h:26,
                 from /home/cc/velox/velox/core/ExpressionEvaluator.h:18,
                 from /home/cc/velox/velox/connectors/Connector.h:27,
                 from /home/cc/velox/velox/connectors/tpch/TpchConnector.h:19,
                 from ten.cc:18:
/home/cc/velox/velox/common/base/SimdUtil.h: In instantiation of ‘uint32_t facebook::velox::simd::crc32U64(uint32_t, uint64_t, const A&) [with A = xsimd::sse2; uint32_t = unsigned int; uint64_t = long unsigned int]’:
/home/cc/velox/velox/exec/VectorHasher.h:78:28:   required from here
/home/cc/velox/velox/common/base/SimdUtil.h:355:43: error: ‘apply’ is not a member of ‘facebook::velox::simd::detail::Crc32<long unsigned int, xsimd::sse2>’
  355 |   return detail::Crc32<uint64_t, A>::apply(checksum, value, arch);
      |          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
```
Where I already build and installed xsimd version 10.0.0 on my Ubuntu 24.04 machine, on an Intel Xeon Ice Lake machine. What did I miss?

**Comment by czentgr** [comment, marked as answer]
@yizhuoliang How would you link the velox library into the executable `ten`? Is it the same source as the
https://github.com/facebookincubator/velox/blob/main/velox/exec/tests/VeloxIn10MinDemo.cpp

Why don't you start building the sample program from the regular build that includes all required libraries and dependencies?

**Reply by yizhuoliang** [reply]
Yeah you are right, I initially tried to build examples outside the velox build itself, but there were too many libs/flags/includes to manage, so later just use the regular build.

**Reply by czentgr** [reply]
No worries. There are efforts and discussions on making a better interface for the Velox API that can then be better leveraged with a shared Velox library.
